### DISCUSSION TITLE
Unauthorized access (missing reservation)

### DISCUSSION THREAD

**Discussion by r-xyz** [usnistgov/NEMO#147] [Ideas]
Hi all,

I was reviewing the `Customizations` menu and noticed that there is a mention of Unauthorized access notification if an user attempts to activate a tool without an existing reservation.
This is mentioned both under [Email addresses](https://github.com/usnistgov/NEMO/blob/b355c62a37d1f6761f8f2e7e66ba284dbe1acf29/NEMO/templates/customizations/customizations_emails.html#L41) and [File and Email templates](https://github.com/usnistgov/NEMO/blob/b355c62a37d1f6761f8f2e7e66ba284dbe1acf29/NEMO/templates/customizations/customizations_templates.html#L334), but the feature appear not to be there (or maybe I am missing how to activate it).
![image](https://github.com/usnistgov/NEMO/assets/100710244/da3b5525-961d-43c2-ae6d-57fd28fcfa92)

Currently the `Unauthorized tool access` event is only fired when an user accesses an area without an existing reservation.
For our use case, it would be interesting to also have the option to make reservation mandatory for tools (either generally or on a one by one case), since we currently do not register area access. Consequently, it would also be interesting to have the alert option.
Since the feature is already mentioned across the program, I was wondering if there was a plan to add it, or if it was considered but then discarded.

**Comment by rptmat57** [comment]
Hi,
Just to clarify, it is sent when a user:
1. tries to activate a tool which has `requires area access` set and the user is not already logged in to the area
2. tries to activate a tool which has `requires area access` set and the area has `reservation required` turned on, and the user is logged in that area but does not have a current reservation for that area (in between reservations, reservation ended a while back, etc.)
Those 2 cases are implemented and working as intended.

You may want to check out [NEMO Community Edition](https://gitlab.com/nemo-community/nemo-ce) which has that [tool reservation required feature ](https://gitlab.com/nemo-community/nemo-ce/-/wikis/Features/Tool-reservation-required-option) you are mentioning, but no `Unauthorized tool access` linked to it.

I could see the benefit of using it in conjunction with the tool reservation required flag and it would make sense to follow the same process (and use the same policy, 15 min grace period, to prevent abuse emails being sent if the user is simply a few minutes early and no one else was using the tool previously).
You could file a ticket in NEMO CE for that if you want

**Comment by r-xyz** [comment]
Hi Mathieu,
Thanks a lot for the feedback. 
I was not aware of the NEMO CE version. To better understand and avoid spamming in the wrong place, I have a few question:

1. Can a current NEMO instance be easily switched to NEMO CE? Will it in the future? I mean, will the DB be kept compatible?
2. Will future Community suggestions only be incorporated in NEMO CE fork and not in NEMO? Should suggestion and PRs be filed directly there?
3. How will NEMO CE be maintained? Will Official NEMO patches and improvements always be backported to NEMO CE in the future?


Regarding my original suggestion, I actually realized the same extra option for tools would also be interesting in `Out of time reservation email` (when a user is still logged to a tool/area but reservation expires).

**Reply by rptmat57** [reply]
Hi,
These are very good questions, see my answers below:
1. Yes, it is, and should be in the future. NEMO CE migrations are kept separate and work similar to app migrations.
2. Yes and no, NEMO will always be the first stop to submit suggestions and discussions. NEMO CE acts as a secondary path if the proposed change or feature cannot/will not be added into NEMO for some reason or other.
3. NEMO is maintained by [NIST](https://www.nist.gov/cnst), NEMO CE is maintained by [Atlantis Labs](https://www.atlantislabs.io/). Every new patch/fix/version of NEMO will be merged into NEMO CE (within a few weeks)

**Reply by r-xyz** [reply]
Great,
Thanks for the reply. I think we could give NEMO CE a try then.
One question: is there any version numbering matching with NEMO? Does NEMO CE 1.5.X goes against NEMO 4.5.X source?
PS: I looked into filing a ticket under NEMO CE, but the issue tab is not visible to normal users (probably active only for project developers).

**Reply by rptmat57** [reply]
Alright, I made the issues tab visible in NEMO CE.
At the moment there is no version numbering matching with NEMO, but I will give it some thought.
